macOS/DuckDuckGo/Menus/MainMenuActions.swift

```diff
@@ -68,7 +68,7 @@ extension AppDelegate {
 
     @objc func newAIChat(_ sender: Any?) {
         DispatchQueue.main.async {
-            AIChatTabOpener.openAIChatTab()
+            AIChatTabOpener.shared.openAIChatTab()
```

Comment:
Let’s not add new singletons. Instead, if you need an app-wide instance of an object, declare it in App Delegate. If you think you’ll bloat App Delegate too much by adding several new properties, consider creating `AIChatCoordinator` that would initialize them all and keep that one in App Delegate (similar to `newTabPageCoordinator` that instantiates NTP actions manager).

These singletons are now small and easy, but in case they grow and we start facing problems due to lack of control over their initialization time, we’ll get into trouble :)

Reply:
Good call. Thanks for the suggestion!